plumbing/transport/file/client.go

```diff
@@ -2,132 +2,134 @@
 package file
 
 import (
-	"bufio"
 	"errors"
 	"io"
 	"os"
-	"path/filepath"
-	"strings"
 
+	"github.com/go-git/go-git/v5/internal/repository"
```

Comment:
As this change would be done within a minor release, let's also keep the previous behaviour (marked as deprecated) within `plumbing/transport/file` (say in `client_git.go`), so that users can revert back to via InstallProtocol - in case our implementation introduces some regression.